### PR TITLE
Port MFPaint

### PR DESCRIPTION
**Pull request by dylwhich**
### Description

Adds MFPaint, ported from last year's 1.1 firmware. Also updates the paint UI to make better use of the touchpad and to not be afraid of un-drawing the canvas sometimes.

### Test Instructions

Known issues:
* The save menu is not yet fully moved into the tool wheel but still partially removed so the old one doesn't work either
* The tutorial overlaps the tool wheel and hasn't really been updated yet
* Using the quick menu breaks things. I'll sort it out eventually.
* I bet very bad things will happen if you manage to swap the palette

Tested on the real swadge and the emulator. Crashes seem to be fixed, and the canvas storing seems solid.

So far, only the main draw screen has been updated. Try out the tool wheel! Draw stuff and see if it breaks.

### Ticket Links

Reference #27 

### Readiness Checklist
- [x] I have run `make format` to format the changes
- [x] I have compiled the firmware and the changes have no warnings
- [x] I have compiled the emulator and the changes have no warnings
- [x] I have run `make cppcheck` and checked that `cppcheck_result.txt` has no warnings for the changes
- [x] I have added doxygen comments to any code used by more than one Swadge mode. This includes `/*! \file` comments with Design Philosophy, Usage, and Example sections for new headers.
- [x] I have run `make docs` and checked that `doxy_warnings.txt` has no warnings for the new code

### CODE REVIEW

File: main/modes/mfpaint/paint_song.c
Comment (AEFeinstein) [inline main/modes/mfpaint/paint_song.c:1]: This should be a midi asset, eventually.

**Comment by dylwhich** [comment]
Ugh, stupid `%zu`s. I'll just... get rid of those...

File: main/modes/mfpaint/wheel_menu.h
Comment (AEFeinstein) [inline main/modes/mfpaint/wheel_menu.h:1]: Wheel to maybe end up in `touchUtils.c` idk

**Review by AEFeinstein — APPROVED** [review]

**Comment by AEFeinstein** [comment]
Uh, I fixed the format specifiers 8c2d95c4cd7940723bcb4d4a185731adbb7b0f09

**Comment by dylwhich** [comment]
Oh, that's why my push failed. Ok! Let me undo that, because I also moved wheelMenu before I saw your comment

**Comment by AEFeinstein** [comment]
I'm also adding some comments for the changes in the generic menu code and will let CI do its thing afterwards

**Review by AEFeinstein — APPROVED** [review]

**Pull request merged**